arbnode/api.go

```diff
@@ -115,21 +117,36 @@ type PricingModelHistory struct {
 	L1PayRewardTo          string     `json:"l1PayRewardTo"`
 }
 
-func (api *ArbDebugAPI) PricingModel(ctx context.Context, start, end rpc.BlockNumber) (PricingModelHistory, error) {
+func (api *ArbDebugAPI) evenlySpaceBlocks(start, end rpc.BlockNumber) (uint64, uint64, uint64, uint64, error) {
 	start, _ = api.blockchain.ClipToPostNitroGenesis(start)
 	end, _ = api.blockchain.ClipToPostNitroGenesis(end)
 
 	blocks := end.Int64() - start.Int64() + 1
-	if blocks > int64(api.blockRangeBound) {
-		log.Warn("Sanitizing pricing model # of blocks", "requested", blocks, "truncated", api.blockRangeBound)
-		blocks = int64(api.blockRangeBound)
+	bound := int64(api.blockRangeBound)
+	step := int64(1)
+	if blocks > bound {
+		step = int64(float64(blocks)/float64(bound) + 0.5)
```

Comment:
This could be written as `(blocks+bound/2)/bound` I think but perhaps the float approach makes it clearer, idk

Reply:
Yeah it's hard to say. I used floats thinking it'd be a tiny bit clearer but the casts are awkward.

Maybe let's keep this as is